swan-lake/development-tutorials/test-document-the-code/test-ballerina-code/execute-tests-in-parallel.md

```diff
@@ -0,0 +1,113 @@
+---
+layout: ballerina-testing-code-left-nav-pages-swanlake
+title: Execute tests in parallel
+description: Learn how to execute tests in parallel.
+keywords: ballerina, programming language, testing, test execution
+permalink: /learn/test-ballerina-code/execute-tests-in-parallel/
+active: execute-tests
+intro: The sections below include information about executing tests in parallel.
+---
+
+## Enable parallel execution of tests
+
+Tests are executed serially by default. User has to consciously enable the parallel execution by passing
+the `--parallel` flag as follows.
+
+```bash
+bal test --parallel
+```
+
+Prior to executing tests concurrently, it is imperative to guarantee concurrency safety. Ballerina inherently assesses the concurrency safety of tests to a certain degree using a predefined set of rules. Tests failing to adhere to these rules are executed serially, notwithstanding the activation of the parallel flag.
```

Comment:
Isn't the proper term "thread safety" here?

Reply:
It seems, we have used the term `concurrency-safe` in our documentation. 
Ref - https://ballerina.io/why-ballerina/concurrent/